### DISCUSSION TITLE
How to enable DHCPv6 in AGH?

### DISCUSSION THREAD

**Discussion by Waldorf3** [AdguardTeam/AdGuardHome#5762] [Q&A]
![image](https://user-images.githubusercontent.com/104714194/233707037-74716ebb-513f-4065-a54f-86c2cdc3d1bc.png)

I have a perfectly fine working setup with AGH serving both DHCP and DNS for IPv4, but after enabling IPv6 on my router, my devices are now getting ISP provided DHCPv6 and DNSv6.  I need IPv6 in order to be able to connect to IPv6-only servers.

AGH > settings > DNS settings > DHCP IPv4 Settings show a normal configuration, but no range can be added under DHCP IPv6 Settings - the fields are grayed out. I have a feeling this could be the problem?

Essentially I just need to know how to enable IPv6 from AGH to see if that will solve the problem. 

EDIT

A little more info. My router is a tp-link Archer C7 v5.0 with DHCPv4 disabled. There are apparently no way to explicitly disable DHCPv6 though. Once IPv6 is enabled there are 3 choices available:  DHCPv6, SLAAC+Stateless DHCP and SLAAC+RDNNS.   I just realized that when using the first two options, clients are assigned the ISP DNS servers, but with the SLAAC+RDNNS option they are assigned the AGH DNS server.  I can't say I understand anything about this, other than this part works.

I still don't see any way to set up DHCPv6 STATIC LEASES in AGH.

**Comment by mainmind83** [comment]
Remember that IPv6 must be enabled in Home Assistant, try in automatic mode to start and restart the AdGuard Home plugin.

![image](https://github.com/user-attachments/assets/8d31ebf7-89ba-4937-8781-abd0ac27f01c)

[DHCP IPv6 6to4](https://support.zyxel.eu/hc/article_attachments/4406708085522/How_to_set_up_6to4_on_the_WAN_and_DHCPv6_on_the_LAN.pdf): 2002:3dde:4b11:1::2

![image](https://github.com/user-attachments/assets/910e10dd-32f7-49d9-aaff-911ceb6e6f01)

DHCP IPV6 private range:

![image](https://github.com/user-attachments/assets/3149b89e-8d47-4064-9560-018393b6893a)


Test on windows with command: 

- ipconfig /release6
- ipconfig /renew6

fd61/64 private range DHCPv6:

- IPv6 on HA: fd61:5447:6b5e:e343::1
- IPV6 start range on AdGuard: fd61:5447:6b5e:e343:0:0:0:2

NOTE: restart system after network change (problems with multicast)

![image](https://github.com/user-attachments/assets/b0e6d3e0-780d-461e-b56b-bb18b291537f)
